src/components/App/App.js

```diff
@@ -7,50 +8,42 @@ import IconButton from "../IconButton/IconButton";
 import DiceRoller from "../DiceRoller/DiceRoller";
 import Footer from "../Footer/Footer";
 import Rules from "../Rules/Rules";
-// Data Models
-import defaultRounds from "../../rounds";
-// Functions
-import {
-  setQueryStringValue,
-  setRoundsUsingQueryString,
-} from "../../utilities/queryString";
-import randomizeMinigames from "../../utilities/randomizeMinigames";
+// Utilities
 import copyLink from "../../utilities/copyLink";
-import toggleRules from "../../utilities/toggleRules";
+import useSetRealms from "../../hooks/useSetRealms";
+import useAreRulesOpen from "../../hooks/useAreRulesOpen";
+import actionCreators from "../../reducers/actionCreators";
 // Assets
 import "./App.scss";
 import printIconSrc from "../../images/print.png";
 import linkIconSrc from "../../images/link.png";
 import randomizeIconSrc from "../../images/randomize.png";
 import rulesIconSrc from "../../images/rules.png";
 
-const App = () => {
-  const [rounds, setRounds] = useState(defaultRounds);
-  const [rulesOpen, setRulesOpen] = useState(false);
-
-  useEffect(() => {
-    setRoundsUsingQueryString(rounds, setRounds);
-  }, []);
-
-  useEffect(() => {
-    setQueryStringValue(rounds);
-  }, [rounds]);
+const useToggleRules = () => {
+  const dispatch = useDispatch();
+  return () => dispatch(actionCreators.toggleRules());
+};
 
-  const updateMinigame = (minigame, round) => {
-    const updatingRoundOrder = { ...rounds };
-    updatingRoundOrder[round] = minigame;
-    setRounds({ ...updatingRoundOrder });
-  };
+const App = () => {
+  const {
+    updateGameOrder,
+    randomizeRealms,
+    realms,
+    setRealms,
+  } = useSetRealms();
+  const toggleRules = useToggleRules();
+  const areRulesOpen = useAreRulesOpen();
 
   return (
     <div className="appContainer">
-      <Rules rulesOpen={rulesOpen} setRulesOpen={setRulesOpen} />
+      <Rules rulesOpen={areRulesOpen} setRulesOpen={toggleRules} />
```

Comment:
Does `Rules` do anything with `setRulesOpen`? I seem to think we decided it does not. If not, let's not pass things in that don't get used.